### ISSUE TITLE
namespace handling

### ISSUE THREAD

**Issue by jennybc** [rsheets/rexcel#13]
`xml2` now makes namespaces easier to deal with. I feel like we should somehow take advantage of this. But am not sure what's best.

Because it has also become clear we have namespace problems, if we want to read unusual sheets like the Ekaterinburg sheet, which has alternative namespaces in many parts of the xml. Perhaps the first thing to decide is whether this is really a goal?

`xml2` good news:
- You can simply stop storing and specifying the namespaces, i.e. `ns` now defaults to `xml_ns()` of the input inside `xml_find_*()`.
- You can strip default namespaces using `xml_ns_strip()`. Now you don't need, e.g., the `d1` prefix.
- You can use `xml_ns_rename()` to provide new prefixes.

If we want to accommodate sheets like Ekaterinburg, we could either detect the alternative namespace and rename them or write XPath that avoids the namespaces. In my experimentation, I've done both.

[Namespace renaming](https://github.com/rsheets/rexcel/blob/d8d62f9eb45931270c245d098806abd556cf5e9e/R/shared_strings.R#L25-L31) re: shared strings

Namespace avoidance example:

```
xml2::xml_find_first(xml, ".//*[local-name() = 'sheets']")
```

**Comment by richfitz** [comment]
Why must XML be so horrible to deal with?

**Comment by jennybc** [comment]
As per Skype, we're going to investigate if Ekaterinburg has truly nonstandard namespaces or just nonstandard prefixes. If the latter, which we suspect, then we should devise some general solution to use in low-level functions that ingest the XML. Here's a current failure trying to read Ekaterinburg. The fatal error happens in `xlsx_ct_sheet()` but I doubt a fix specific to each function is the way to go.

```
3. Error: Ekaterinburg_IP_9.xlsx (@test-rexcel-read-workbook.R#18) -------------
deal with me
1: expect_silent(rexcel_read_workbook(sh)) at /Users/jenny/rrr/rexcel/tests/testthat/test-rexcel-read-workbook.R:18
2: evaluate_promise(object)
3: with_sink(temp, withCallingHandlers(withVisible(code), warning = handle_warning, 
       message = handle_message))
4: withCallingHandlers(withVisible(code), warning = handle_warning, message = handle_message)
5: withVisible(code)
6: rexcel_read_workbook(sh)
7: xlsx_read_workbook(path) at /Users/jenny/rrr/rexcel/R/read.R:39
8: xlsx_ct_sheets(xml, ns, rels) at /Users/jenny/rrr/rexcel/R/read_workbook.R:8
9: process_container(xml, "d1:sheets", ns, xlsx_ct_sheet) at /Users/jenny/rrr/rexcel/R/read_workbook.R:15
10: rbind_df(lapply(els, fun, ns, ...), classes) at /Users/jenny/rrr/rexcel/R/utils.R:43
11: tibble_empty_data_frame(classes) at /Users/jenny/rrr/rexcel/R/utils.R:55
12: stop("deal with me") at /Users/jenny/rrr/rexcel/R/utils.R:75
```

**Comment by richfitz** [comment]
Brilliant, thanks. Will look at this tomorrow morning.

**Comment by richfitz** [comment]
in progress (offline)

**Comment by richfitz** [comment]
Namespacing should have been fixed a in f577e63
